core/Updates/5.0.0-b1.php

```diff
@@ -86,28 +85,34 @@ private function getUpdateArchiveIndexMigrations()
         return $migrations;
     }
 
-    private function getLogVisitTableMigrations($migrations)
+    private function appendLogVisitTableMigrations($migrations)
     {
-        $migrations[] = $this->migration->db->dropIndex('log_visit', $this->indexName);
-
-        // Using the custom `sql` method instead of the `addIndex` method as it doesn't support DESC collation
-        $migrations[] = $this->migration->db->sql(
-            "ALTER TABLE `{$this->tableName}` ADD INDEX `{$this->indexName}` (`idsite`, `idvisitor`, `visit_last_action_time` DESC)",
-            [DbAlias::ERROR_CODE_DUPLICATE_KEY, DbAlias::ERROR_CODE_KEY_COLUMN_NOT_EXISTS]
-        );
+        if ($this->hasNewIndex()) {
+            // correct index already exists, so don't perform anything
+        } else if ($this->hasCorrectlySetOldIndex() && $this->doesMySQLSupportRenameIndex()) {
```

Comment:
@michalkleiner @sgiehl @bx80 would this for example cause an issue on MariaDB 10.0 where it's not supported yet? https://mariadb.com/kb/en/alter-table/#rename-indexkey 
Maybe this edge case we wouldn't need to treat separately and better to just always delete and add in order to rename? Might be unclear what other edge cases there may be.

Reply:
I've now created https://github.com/matomo-org/matomo/pull/20884 to handle maria db when checking if an index rename is possible.
We could for sure also always recreate the index, but that would actually kill the benefit we tried to create with having the new index already in place for new installs with one of the Matomo 4 releases. 
But yes if try to rename the index on a system that doesn't support it, that might cause other trouble now. Do we know how many people are using such outdated database versions?

Reply:
@sgiehl @michalkleiner could we maybe rather feature detect it? Usually, checking for version could be causing all sorts of problems as the versions may not report reliably or for example not contain `mariadb` or there might be other MySQL like DBs that will have issues with it.

Reply:
@tsteur not sure how to do a feature detection in mysql. Anyway, I have updated the PR, so it doesn't check for mysql versions. It will always try to rename the index if the old one is already correct. Afterwards if tries to create the new one and removes the old one. Depending on the availability to rename index support either the first query will fail or the latter ones. Possible errors are marked as ignored, so this imho should work everywhere.